lm_eval/tasks/evalita_llm/utils.py

```diff
@@ -0,0 +1,576 @@
+    return ans_str
+
+
+def ls_doc_to_target(x):
+    """
+    Generate the target for the lexical similarity task
+    """
+    if len(x["answers"]) == 0:
+        return NO_SYN_STRING
+    ans_str = ""
+    for i in x["answers"]:
+        ans_str += i["word"] + ", "
+    if len(ans_str) != 0 and ans_str[-2] == ",":
+        ans_str = ans_str[:-2]
+    return ans_str
+
+
+def _ls_split_gold(x):
+    """
+    Split the gold string into a list of tuples
+    """
+    if x == NO_SYN_STRING:
+        return [], []
+    answers = x.split("::")
+    words = []
+    freqs = []
+    if len(answers) != 0:
+        for a in answers:
+            if "$$" in a:
+                word, count = a.split("$$")
+                words.append(word)
+                try:
+                    freqs.append(int(count))
+                except ValueError:
+                    freqs.append(0)
+    return words, freqs
+
+
+def ls_process_results(doc, results):
+    """
+    Process the results of the evaluation for the lexical substitution task
+    look at coqa for another example
+    """
+    gold_to_target = _ls_gold_to_target(doc)
+    words, freqs = _ls_split_gold(gold_to_target)
+    prec = 0
+
+    # Considering a maximum of the first 10 synonyms
+    results = split_text_with_regex(results[0], LS_SPLIT_REGEX)
+    results = results[: min(10, len(results))]
+
+    # Remove non-alphabetic characters from the word at the end of the list
+    if results:  # Check if results is not empty
+        results[-1] = "".join(char for char in results[-1] if char.isalpha())
+
+    has_answ = 0 if len(results) == 0 else 1  # so we can compute |A|
+    has_annotation = 0 if len(words) == 0 else 1  # so we can compute |T|
+
+    matching_res = []  # for debugging
+
+    for r in results:
+        if r in words:
+            # get frequency of the synonyms from annotators
+            idx = words.index(r.strip())
+            prec += freqs[idx]
+            matching_res.append(r)
+
+    # In the case of the OOT (out of ten) subtask, this normalization should not be applied
+    # ai = len(results) if len(results) != 0 else 1
+    # prec = prec / ai
+
+    Hi = sum(freqs)
+    if Hi != 0:
+        prec = prec / Hi
+    else:
+        eval_logger.debug("H_i is 0")
+
+    return {"f1": (prec, has_answ, has_annotation)}
+
+
+# ---------------------- NER ----------------------
+
+NO_ENT_STRING = "&&NOENT&&"
+NER_ENTITY_SEPARATOR = ","
+NER_TYPE_SEPARATOR = "$"
+NER_MAPPING_V2 = {"PER": 0, "LOC": 1, "ORG": 2, NO_ENT_STRING: 3, "O": 4}
+NER_MAPPING = {"PER": 0, "LOC": 1, "ORG": 2, "O": 3}
+
+
+def _ner_gold_to_target(x: list) -> list:
+    """
+    Convert the gold entities to the target format according to the NER_MAPPING
+    """
+    res = [NER_MAPPING[e["type"]] for e in x]
+    return res
+
+
+def _ner_gold_to_target_v2(x: list) -> list:
+    """
+    Convert the gold entities to the target format according to the NER_MAPPING
+    """
+    res = [NER_MAPPING[e["type"]] for e in x]
```

Comment:
Hey @m-resta, is this supposed to be `NER_MAPPING_V2`? the `NR_MAPPING_V2` dict is unused, and this function is identical to the above one.